wikidatarefisland/data_access/storage.py

```diff
@@ -34,7 +34,9 @@ def append(self, file_, value, raw=False):
             if raw:
                 f.write(value)
             else:
-                f.write(json.dumps(value))
+                # Add newline if file is not empty
+                new_line = '\n' if os.stat(f.name).st_size else ''
```

Comment:
Can it be something like f.write(json.dumps(value) + '\n')? You will have an extra newline at the end of the file but the logic is much simpler

Reply:
That was what I initially did, but then realised that `.readLines()` would then go over this empty line too, which is not ideal. Also I think it's cleaner if you don't add any extra newlines to a json (or jsonl) file. How about i put this in it's own private method like this:

```py
def _is_empty(file_):
    return os.stat(file_.name).st_size == 0
```

and then:

```py
new_line = '\n' if not _is_empty(f) else ''
```

Reply:
nah, I think it's fine. not a big deal